### PR TITLE
Add ActivateVolumeWithMultipleTPMSealedKeys

### PR DESCRIPTION
**Pull request by chrisccoulson**
This adds the ability to activate a volume with multiple TPM sealed key objects. The existing API is not just a wrapper around the new API.

I'm not sure what to do about returning errors from this function - I've added a new ActivateWithMultipleTPMSealedKeysError which has the ability to access each of the individual errors, but I'm not sure that it's actually worth surfacing all of this information.

I've renamed RecoveryKeyUsageReason to KeyErrorCode, as its usage is now more associated with why a key failed rather than why the recovery key was used. Where multiple keys are supplied and they all fail, they each have their own error code.

### CODE REVIEW

**Review by pedronis — COMMENTED** [review]
this looks reasonable,  error handling can maybe be simplified a bit, also the multi-key error formatting doesn't quite fit with snapd usual patterns.  @chrisccoulson I will push the necessary tweaks myself.

**Review by pedronis — APPROVED** [review]
thanks, I also applied the tweaks from my own feedback

File: crypt.go [hunk 1]
```diff
@@ -584,12 +583,7 @@ func ActivateVolumeWithMultipleTPMSealedKeys(tpm *TPMConnection, volumeName, sou
 		return false, err
 	}
 
-	if err := activateWithTPMKeys(tpm, volumeName, sourceDevicePath, keyPaths, passphraseReader, options.PassphraseTries, activateOptions, options.KeyringPrefix); err != nil {
-		errs, ok := err.(activateWithTPMKeysErrors)
-		if !ok {
-			return false, xerrors.Errorf("unexpected error: %w", err)
-		}
-
+	if success, errs := activateWithTPMKeys(tpm, volumeName, sourceDevicePath, keyPaths, passphraseReader, options.PassphraseTries, activateOptions, options.KeyringPrefix); !success {
```
Comment (mvo5) [inline crypt.go:586]: I think the change here is good but the function return still does not feel super natural/go-ish. But no idea what else could be done to improve so +1 :)

**Review by mvo5 — APPROVED** [review]
Thank you!

**Review by bboozzoo — APPROVED** [review]
LGTM

**Pull request merged**